### PR TITLE
add --blob-type for `az storage copy` command

### PR DESCRIPTION
**Pull request by Juliehzl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -389,6 +389,8 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
                     recursively when uploading from local file system.')
         c.argument('put_md5', action='store_true', help='Create an MD5 hash of each file, and save the hash \
                     as the Content-MD5 property of the destination blob/file.Only available when uploading.')
+        c.argument('blob_type', validator=validate_blob_type,
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: Any difference between get_enum_type(get_blob_types())) and it?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: enum values are different. For azcopy, blob types must be one of "BlockBlob", "PageBlob", "AppendBlob".

File: src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_azcopy_scenarios.py [hunk 2]
```diff
@@ -265,6 +265,16 @@ def test_storage_azcopy_blob_url(self, resource_group, first_account, second_acc
         self.cmd('storage blob list -c {} --account-name {}'
                  .format(first_container, second_account), checks=JMESPathCheck('length(@)', 21))
 
+        # Upload to managed disk
+        diskname = self.create_random_name(prefix='disk', length=12)
+        local_file = self.create_temp_file(1024)
+        self.cmd('az disk create -n {disk} -g {rg} --for-upload -z 10'
+                 .format(diskname, resource_group))
+        sasURL = self.cmd('az disk grant-access --access-level Write --duration-in-seconds 3600 -n {} -g {} --query [accessSas] -o tsv'
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_azcopy_scenarios.py]: nit-pick: `--query accessSas` should be enough.

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 3]
```diff
@@ -389,6 +389,8 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
                     recursively when uploading from local file system.')
         c.argument('put_md5', action='store_true', help='Create an MD5 hash of each file, and save the hash \
                     as the Content-MD5 property of the destination blob/file.Only available when uploading.')
+        c.argument('blob_type', arg_type=get_enum_type(["BlockBlob", "PageBlob", "AppendBlob"]),
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: Is the enum type defined in Python SDK? 
E.g. https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-mgmt-storage/azure/mgmt/storage/v2019_04_01/models/_storage_management_client_enums.py

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: No. Because it is not a popular type format, I just defined here.

**Pull request merged**